client/src/charts/nivo/nivo_bar.js

```diff
@@ -0,0 +1,273 @@
+    );
+
+    // have to have an empty string in key to make sure that negative bars will be displayed
+    const graph = 
+      <div style={{height: graph_height }} aria-hidden='true'>
+        <ResponsiveBar
+          {...{data,
+            margin,
+            colors,
+            groupMode,
+            enableGridX,
+            enableGridY,
+            colorBy,
+            theme, 
+            indexBy, 
+            onMouseEnter,
+            onMouseLeave,
+            onClick,
+            padding,
+            tooltip,
+            enableLabel,
+            label,
+            borderWidth,
+            isInteractive,
+            animate,
+            motionDamping,
+            motionStiffness,
+          }}
+          legends={ IE_fixed_legends }
+          keys={_.union([''],keys)}
+          labelFormat={_.isUndefined(label_format) ? null : label_format}
+          labelTextColor={window.infobase_color_constants.textColor}
+          tooltip={ (d) => tooltip( [d], get_formatter(is_money, text_formatter, false) ) }
+          axisBottom={remove_bottom_axis ? null : bttm_axis}
+          axisLeft={
+            remove_left_axis ?
+              null :
+              {
+                tickValues: 6,
+                format: (d) => get_formatter(is_money, text_formatter)(d),
+                min: "auto",
+                max: "auto",
+                ...(left_axis || {}),
+              }
+          }
+        />
+      </div>;
+    
+    return <InteractiveGraph graph={graph} table={table} table_name={table_name}/>;
+  }
+};
+NivoResponsiveBar.defaultProps = {
+  ...general_default_props,
+  padding: 0.3,
+  bttm_axis: {
+    tickSize: 7,
+    tickPadding: 10,
+    tickRotation: 0,
+  },
+};
+  
+  
+  
+export class NivoResponsiveHBar extends React.Component{
+  render(){
+    const{
+      data,
+      keys,
+      groupMode,
+      indexBy,
+      margin,
+      legends,
+      graph_height,
+      bttm_axis,
+      left_axis,
+      top_axis,
+      remove_bottom_axis,
+      remove_left_axis,
+      add_top_axis,
+      enableLabel,
+      label,
+      label_format,
+      labelSkipWidth,
+      is_money,
+      text_formatter,
+      theme,
+      colors,
+      colorBy,
+      tooltip,
+      enableGridX,
+      enableGridY,
+      padding,
+      markers,
+      disable_table_view,
+      table_name,
+      table_first_column_name,
+      isInteractive,
+      motionDamping,
+      motionStiffness,
+    } = this.props;
+    
+    const IE_fixed_legends = legends ? (
+      _.map(legends,
+        legend => _.chain(legend)
+          .clone()
+          .assign({symbolShape: fixed_symbol_shape})
+          .value() )
+      ) :
+      undefined;
+
+    const table = !disable_table_view && bar_table(data, keys, indexBy, get_formatter(is_money, text_formatter, true, true), table_name, table_first_column_name);
+
```

Comment:
Is this the same code from lines 81-92? Please de-dupe if so, so the logic doesn't become inconsistent later by accident.